tests/_utils/view.js

```diff
@@ -0,0 +1,836 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/* jshint latedef:false */
+
+import ViewDocumentFragment from '/ckeditor5/engine/treeview/documentfragment.js';
+import HtmlDataProcessor from '/ckeditor5/engine/dataprocessor/htmldataprocessor.js';
+import ViewElement from '/ckeditor5/engine/treeview/element.js';
+import Selection from '/ckeditor5/engine/treeview/selection.js';
+import Range from '/ckeditor5/engine/treeview/range.js';
+import Position from '/ckeditor5/engine/treeview/position.js';
+import AttributeElement from '/ckeditor5/engine/treeview/attributeelement.js';
+import ContainerElement from '/ckeditor5/engine/treeview/containerelement.js';
+import ViewText from '/ckeditor5/engine/treeview/text.js';
+
+const DomDocumentFragment = window.DocumentFragment;
+const DomElement = window.Element;
+
+const ELEMENT_RANGE_START_TOKEN = '[';
+const ELEMENT_RANGE_END_TOKEN = ']';
+const TEXT_RANGE_START_TOKEN = '{';
+const TEXT_RANGE_END_TOKEN = '}';
+
+/**
+ * Converts view elements to its HTML-like string representation.
+ * Root element can be provided as {@link engine.treeView.Text Text}:
+ *
+ *		const text = new Text( 'foobar' );
+ *		stringify( text ); // 'foobar'
+ *
+ * or as {@link engine.treeView.Element Element}:
+ *
+ *		const element = new Element( 'p', null, new Text( 'foobar' ) );
+ *		stringify( element ); // '<p>foobar</p>'
+ *
+ * or as {@link engine.treeView.DocumentFragment DocumentFragment}:
+ *
+ *		const text = new Text( 'foobar' );
+ *		const b = new Element( 'b', { name: 'test' }, text );
+ *		const p = new Element( 'p', { style: 'color:red;' } );
+ *		const fragment = new DocumentFragment( [ p, b ] );
+ *
+ *		stringify( fragment ); // '<p style="color:red;"></p><b name="test">foobar</b>'
+ *
+ * Additionally {@link engine.treeView.Selection Selection} instance can be provided, then ranges from that selection
+ * will be included in output data.
+ * If range position is placed inside element node, it will be represented with `[` and `]`:
+ *
+ *		const text = new Text( 'foobar' );
+ *		const b = new Element( 'b', null, text );
+ *		const p = new Element( 'p', null, b );
+ *		const selection = new Selection();
+ *		selection.addRange( Range.createFromParentsAndOffsets( p, 0, p, 1 ) );
+ *
+ *		stringify( p, selection ); // <p>[<b>foobar</b>]</p>
```

Comment:
Is it possible to pass range to that method directly, without creating a selection? Could I pass position as well?